### PR TITLE
minimum number od jumps solution added

### PR DESCRIPTION
**Pull request by Yashgabani845**
## Related Issue
Fixes : #518 

## Description
Added solution of the problem in full deatils 
more than 350 lines of code

1. code in cpp,java,python
2. brute force approch
3. optimal brute force approch 
4. code for every approch
5. expected approch
6. complexity analysis
7. conclusion

## Type of PR

- [x] Feature enhancement
- [x] Documentation update

## Screenshots / Videos (if applicable)

Screenshot (422)
Screenshot (421) and Screenshot (420)
## Checklist
- [ ] I have performed a self-review of my code.
- [ ] I have read and followed the Contribution Guidelines.
- [ ] I have tested the changes thoroughly before submitting this pull request.
- [ ] I have provided relevant issue numbers, screenshots, and videos after making the changes.
- [ ] I have commented my code, particularly in hard-to-understand areas.

## Additional Context

thanks for this opprtunity kindly go through file changes and if needed give the changes
i will ahppy to improve and learn

## Resources for Guidance

Here are some resources that may be helpful as you contribute to CodeHarborHub:
- [Code Harbor Hub Community Features](https://www.codeharborhub.live/community/features)
- [Docusaurus Documentation](https://docusaurus.io/docs/create-doc)
- [React.js Documentation](https://legacy.reactjs.org/docs/getting-started.html)
- [Markdown Guide](https://www.markdownguide.org/)
- [MDX Documentation](https://mdxjs.com/docs/)
- [Mermaid Documentation](https://mermaid.js.org/)

**Resources for DSA Features on Our Site:**

- Tabs: [Switch Languages](https://www.codeharborhub.live/community/features#multi-language-support-code-blocks)
- [Line Numbering](https://www.codeharborhub.live/community/features#line-numbering)
- [Math Equations](https://www.codeharborhub.live/community/features#math-equations-with-katex)

### CODE REVIEW

**Comment by Yashgabani845** [comment]
@Ajay-Dhangar done with all works you assigned me yesterday

File: dsa-solutions/gfg-solutions/0022-minimum-number-of-jumps.md [hunk 1]
```diff
@@ -0,0 +1,356 @@
+---
+id: minimum number of jumps
```
Comment (Ajay-Dhangar) [inline dsa-solutions/gfg-solutions/0022-minimum-number-of-jumps.md:2]: replace `id: minimum number of jumps` to `id: minimum-number-of-jumps`

File: dsa-solutions/gfg-solutions/0022-minimum-number-of-jumps.md [hunk 2]
```diff
@@ -0,0 +1,356 @@
...
+#### Intuition
+
+The brute force approach involves recursively checking all possible jumps from each element, calculating the minimum number of jumps needed to reach the end from reachable elements.
+
+#### Implementation
+
+The implementation recursively checks all possible jumps from each element, updating the minimum number of jumps needed.
+
+```cpp
+// C++ Implementation
```
Comment (Ajay-Dhangar) [inline dsa-solutions/gfg-solutions/0022-minimum-number-of-jumps.md:43]: remove comment `// C++ Implementation`

File: dsa-solutions/gfg-solutions/0022-minimum-number-of-jumps.md [hunk 3]
```diff
@@ -0,0 +1,356 @@
...
+#### Intuition
+
+In this approach, we use dynamic programming to store the minimum number of jumps required to reach each index from the start.
+
+#### Implementation
+
+The code initializes an array to store the minimum jumps required and iterates through the array to calculate the minimum jumps.
+
+```cpp
+// C++ Implementation
```
Comment (Ajay-Dhangar) [inline dsa-solutions/gfg-solutions/0022-minimum-number-of-jumps.md:74]: remove comment `// C++ Implementation`

File: dsa-solutions/gfg-solutions/0022-minimum-number-of-jumps.md [hunk 4]
```diff
@@ -0,0 +1,356 @@
...
+        }
+        return (jumps[n - 1] == INT_MAX ? -1 : jumps[n-1]);
+    }
+};
+```
+
+#### Python Implementation
+
+```python
+# Python Implementation
```
Comment (Ajay-Dhangar) [inline dsa-solutions/gfg-solutions/0022-minimum-number-of-jumps.md:100]: remove comments

**Review by Ajay-Dhangar — CHANGES_REQUESTED** [review]
You can try to remove comments and explain code may be use mermaid graph (like flowchart), and may be used tip for good practice

**Comment by Yashgabani845** [comment]
@Ajay-Dhangar done
removed all comments of cpp implementaiaon, java,python
and also will use flowcharts and graph for next tim instead of using comments 
thanks for your feedback

**Review by Ajay-Dhangar — APPROVED** [review]
Done

**Pull request merged**